### PR TITLE
docs - new topic, updated ref pages for progress reporting views and some utilities

### PR DESCRIPTION
**Pull request by lisakowen**
docs for various progress reporting-related PRs, including but not limited to: https://github.com/greenplum-db/gpdb/pull/14994, https://github.com/greenplum-db/gpdb/pull/15042, https://github.com/greenplum-db/gpdb/pull/15121, https://github.com/greenplum-db/gpdb/pull/15327

in this PR:
- add reference info for `gp_stat_progress_<command>` and `gp_stat_progress_<command>_summary` views, where `<command>` is analyze, create_index, copy, cluster, basebackup, vacuum.
- add new "monitoring long-running operations" topic based on https://www.postgresql.org/docs/15/progress-reporting.html.  includes the info about the progress phases, and calls out any AO/CO table differences.
- add new utility reference page for pg_checksums
- opportunisticly update utility reference pages for clusterdb and reindexdb to bring them up to postgres v12.

NOTE:  i am not sure if i am aware of, or covered, all of the differences between heap AO/CO tables. 

PLEASE:  search for XXX in the modified files - please help fill in this content, or, where it prefaces content, let me know if the text still applicable in greenplum 7.

QUESTIONS:
1.  are there any limitations or known issues to call out?
2.  ok to keep the term term "heap" in phase descriptions that apply to both heap and AO/CO tables?
3.  do we support progress on both partitioned and non-partitioned tables?  and do we support progress in the same way for both?
4.  would a user invoke pg_checksums directly, or indirectly via a utility command (like gprecover seg and basebackup)?

doc review site links (behind vpn):
- progress reporting system views:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/prog-rpt/greenplum-database/GUID-ref_guide-system_catalogs-catalog_ref-views.html#progress-reporting-views
- new "monitoring long-running operations" topic:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/prog-rpt/greenplum-database/GUID-admin_guide-managing-progress_reporting.html
- new pg_checksums utility ref page:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/prog-rpt/greenplum-database/GUID-utility_guide-ref-pg_checksums.html

### CODE REVIEW

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# Monitoring Long-Running Operations
+
+Greenplum Database can report the progress of `ANALYZE`, `CLUSTER`, `COPY`, `CREATE INDEX`, `REINDEX`, and `VACUUM` commands during command execution, with some caveats. Greeplum can also report the progress of a running base backup (initiated during [gprecoverseg](../../utility_guide/ref/gprecoverseg.html)) or [pg_checksums](../../utility_guide/ref/pg_checksums.html) command invocation, allowing you to monitor the progress of these possibly long-running operations.
+
+Greenplum reports the command progress via ephemeral system views. That is, the views return data only while the operations are running. Two sets of progress reporting views are provided:
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:5]: Greenplum reports the command progress via ephemeral system views. That is, the views return data only while the operations are running.  ==>

Greenplum reports the command progress via ephemeral system views, which return data only while the operations are running.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
+# Monitoring Long-Running Operations
+
+Greenplum Database can report the progress of `ANALYZE`, `CLUSTER`, `COPY`, `CREATE INDEX`, `REINDEX`, and `VACUUM` commands during command execution, with some caveats. Greeplum can also report the progress of a running base backup (initiated during [gprecoverseg](../../utility_guide/ref/gprecoverseg.html)) or [pg_checksums](../../utility_guide/ref/pg_checksums.html) command invocation, allowing you to monitor the progress of these possibly long-running operations.
```
Comment (Chibin) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:3]: s/Greeplum/Greenplum

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+- `gp_stat_progress_<command>_summary` - aggregates `<command>` progress on the coordinator and all segments, and displays one row per running `<command>` invocation
+
+Greeplum reports progress in phases, where the phases are specific to the command. For example, `acquiring sample rows` is an analyze progress phase, while `building index` is an index creation progress phase. Greenplum reports the progress for both heap and AO/CO tables. For most commands, heap and AO/CO table progress is reported together. For vacuum and cluster operations, Greenplum reports heap and AO/CO table progress in separate phases.
+
+
+## <a id="analyze_progress"></a>ANALYZE Progress Reporting
+
+The [gp_stat_progress_analyze](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_analyze) system view reports the progress of running `ANALYZE` and `analyzedb` operations. The view displays a row per segment instance that is currently servicing an analyze operation.
+
+For each active analyze operation, the `gp_stat_progress_analyze_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_analyze`, and shares the same schema as this view.
```
Comment (Chibin) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:17]: Just for my curiosity, I noticed that we use Greenplum and sometimes Greenplum Database, is there a difference? and when do we decide to use one or the other?

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+| `swapping relation files` | The command is currently swapping newly-built files into place. |
+| `performing final cleanup` | The command is performing final cleanup. When this phase is completed, `CLUSTER` or `VACUUM FULL` ends. |
+
+## <a id="copy_progress"></a>COPY Progress Reporting
+
+The [gp_stat_progress_copy](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_copy) system view reports the progress of running `COPY` operations. The view displays a row per segment instance that is currently servicing a copy operation.
+
+For each active copy operation, the `gp_stat_progress_copy_summary` aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_copy`, and shares the same schema as this view.
+
+You can also use these views to monitor the data movement progress of utilities that use `COPY` under the hood, such as XXX.
```
Comment (Chibin) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:72]: gpbackup or gprestore  could work for the XXX.

File: gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md [hunk 5]
```diff
@@ -16,63 +16,98 @@ clusterdb -V | --version
 
 ## <a id="section3"></a>Description 
 
-To cluster a table means to physically reorder a table on disk according to an index so that index scan operations can access data on disk in a somewhat sequential order, thereby improving index seek performance for queries that use that index.
+XXX To cluster a table means to physically reorder a table on disk according to an index so that index scan operations can access data on disk in a somewhat sequential order, thereby improving index seek performance for queries that use that index.
```
Comment (l-wang) [inline gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md:19]: The description is correct. The wording: "somewhat sequential order" is true but sounds vague. Maybe we can refer to upstream's description: 
> Clustering is a one-time operation: when the table is subsequently updated, the changes are not clustered. That is, no attempt is made to store new or updated rows according to their index order. 

https://www.postgresql.org/docs/12/sql-cluster.html

File: gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md [hunk 6]
```diff
@@ -16,63 +16,98 @@ clusterdb -V | --version
...
 
-`clusterdb` is a wrapper around the SQL command `CLUSTER`. Although clustering a table in this way is supported in Greenplum Database, it is not recommended because the `CLUSTER` operation itself is extremely slow.
+`clusterdb` is a wrapper around the SQL command [CLUSTER](../../ref_guide/sql_commands/CLUSTER.html). There is no effective difference between clustering databases via this utility and via other methods for accessing the server.
 
-If you do need to order a table in this way to improve your query performance, use a `CREATE TABLE AS` statement to reorder the table on disk rather than using `CLUSTER`. If you do 'cluster' a table in this way, then `clusterdb` would not be relevant.
+XXX If you do need to order a table in this way to improve your query performance, use a `CREATE TABLE AS` statement to reorder the table on disk rather than using `CLUSTER`. If you do 'cluster' a table in this way, then `clusterdb` would not be relevant.
```
Comment (l-wang) [inline gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md:25]: I don't see the purpose of this paragraph. Specifically, 
> If you do need to order a table in this way

What way is this way? It seems missing contaxt. "CREATE TABLE AS" simply creates a new table with same schema and insert same data as old table. It wouldn't reorder the old table. I think this paragraph is confusing and I'm fine with removing it.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+| `sorting tuples` | `CLUSTER` is currently sorting tuples. |
+| `writing new append-optimized` | `CLUSTER` is currently writing the new AO/CO table. |
+| `swapping relation files` | The command is currently swapping newly-built files into place. |
+| `performing final cleanup` | The command is performing final cleanup. When this phase is completed, `CLUSTER` or `VACUUM FULL` ends. |
+
+## <a id="copy_progress"></a>COPY Progress Reporting
+
+The [gp_stat_progress_copy](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_copy) system view reports the progress of running `COPY` operations. The view displays a row per segment instance that is currently servicing a copy operation.
+
+For each active copy operation, the `gp_stat_progress_copy_summary` aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_copy`, and shares the same schema as this view.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:70]: gp_stat_progress_copy_summary view shares same schema as the upstream pg_stat_progress_copy view.
gp_stat_progress_copy view has an additional gp_segment_id column which tells the content id of the particular segment.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+Greenplum Database calculates the `bytes_processed`, `bytes_total`, `tuples_processed`, and `tuples_excluded` column values differently for the `gp_stat_progress_copy_summary` view depending on the type of table and type of `COPY` operation. The table below identifies the table types, the types of `COPY` operations, and which of `sum()` or `average()` Greenplum Database uses to calculate the final value:
+
+| Table type |COPY TO | COPY FROM | COPY TO/FROM ON SEGMENT |
+|-----|-----------|------|------|
+| Distributed table | `sum()` | `sum()` | `sum()` |
+| Replicated table | `sum()` | `average()` | `sum()` |
+
+Greenplum uses `sum()` for `COPY ... ON SEGMENT` as the command explicitly operates on each segment irrespective of the table type.
+
+Greenplum uses `sum()` for `COPY TO` with replicated tables, as the actual copy originates from only a single segment.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:83]: As someone who always have hard time explaining all these caveats with the COPY summary view, I really like your explanations!

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+
+## <a id="create_index_progress"></a>CREATE INDEX Progress Reporting
+
+The [gp_stat_progress_create_index](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_create_index) system view reports the progress of running `CREATE INDEX` and `REINDEX` operations. The view displays a row per segment instance that is currently servicing either command.
+
+For each active index operation, the `gp_stat_progress_create_index_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_create_index`, and shares the same schema as this view.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:90]: gp_stat_progress_create_index_summary view shares same schema as the upstream pg_stat_progress_create_indexview.
gp_stat_progress_create_index view has an additional gp_segment_id column which tells the content id of the particular segment.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+| `index validation: scanning table` | This phase is not applicable, Greenplum Database does not support `CREATE INDEX CONCURRENTLY` or `REINDEX CONCURRENTLY`. |
+| `waiting for old snapshots` | This phase is not applicable, Greenplum Database does not support `CREATE INDEX CONCURRENTLY` or `REINDEX CONCURRENTLY`. |
+| `waiting for readers before marking dead` | This phase is not applicable, Greenplum Database does not support `REINDEX CONCURRENTLY`. |
+| `waiting for readers before dropping` | This phase is not applicable, Greenplum Database does not support `REINDEX CONCURRENTLY`. |
+
+Greenplum Database skips several phases because it does not support concurrent index creation or concurrent reindexing.
+
+## <a id="vacuum_progress"></a>VACUUM Progress Reporting
+
+The [gp_stat_progress_vacuum](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_vacuum) system view reports the progress of running `VACUUM` and `vacuumdb` operations. The view displays a row per segment instance that is currently servicing a vacuum operation.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:111]: as well as VACUUM FULL for AO/CO tables.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+
+For each active vacuum operation, the `gp_stat_progress_vacuum_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_vacuum`, and shares the same schema as this view.
+
+(Regular `VACUUM` modifies the table in place. `VACUUM FULL` rewrites the table as does a `CLUSTER` operation. For information about progress reporting for `VACUUM FULL`, see [CLUSTER and VACUUM FULL Progress Reporting](#cluster_progress).)
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:115]: This is only true for heap tables. See my previous comment.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+In Greenplum Database, an AO/CO table vacuum behaves differently than a heap table vacuum. Because Greenplum stores the logical EOF for each segment file, it does not need to scan physical blocks after the logical EOF, so Greenplum can truncate them. Due to this difference, Greenplum reports vacuum progress on heap and AO/CO tables using different phases.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:117]: In addition, AO/CO always rewrite data into new segment files to get rid of dead tuples, so VACUUM FULL and VACUUM for AO/CO tables perform the same way and there is no "lazy" vacuum for AO/AOCO tables.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+| `waiting for readers before marking dead` | This phase is not applicable, Greenplum Database does not support `REINDEX CONCURRENTLY`. |
+| `waiting for readers before dropping` | This phase is not applicable, Greenplum Database does not support `REINDEX CONCURRENTLY`. |
+
+Greenplum Database skips several phases because it does not support concurrent index creation or concurrent reindexing.
+
+## <a id="vacuum_progress"></a>VACUUM Progress Reporting
+
+The [gp_stat_progress_vacuum](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_vacuum) system view reports the progress of running `VACUUM` and `vacuumdb` operations. The view displays a row per segment instance that is currently servicing a vacuum operation.
+
+For each active vacuum operation, the `gp_stat_progress_vacuum_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_vacuum`, and shares the same schema as this view.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:113]: gp_stat_progress_vacuum_summary view shares same schema as the upstream pg_stat_progress_vacuum view.
gp_stat_progress_vacuum view has an additional gp_segment_id column which tells the content id of the particular segment.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+
+### <a id="vacuum_progress_ao"></a>AO/CO Table Vacuum Phases
+
+The table below describes how to interpret the *AO/CO table* phase-specific information reported in the views:
+
+|AO/CO Table Phase|Description|
+|-----|-----------|
+| `initializing` | `VACUUM` is preparing to begin scanning AO/CO tables. This phase is expected to be very brief. |
+| `vacuuming indexes` | For AO/CO tables, the `vacuuming indexes` phase is a sub-phase that may occur during both `append-optimized pre-cleanup` and `append-optimized post-cleanup` phases if the relation has an index and there are invisible awaiting-drop segment files. |
+| `append-optimized pre-cleanup` | XXX |
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:141]: Something like:
`VACUUM` is currently performing pre-cleanup of the AO/CO table, which includes recycling AWAITING_DROP segments that are no longer visible to anyone from previous VACUUM, and truncating all live segment files to their logcial EOFs.

**Review by l-wang — COMMENTED** [review]
I'm half way.

**Review by l-wang — CHANGES_REQUESTED** [review]
Thank you @lisakowen for puting these together! I've finished my first round of review.

**Comment by l-wang** [comment]
> NOTE: i am not sure if i am aware of, or covered, all of the differences between heap AO/CO tables.
You did great job! Minor things see my individual comments. 

> 1. are there any limitations or known issues to call out?
I need to think about this too and will come back to you.

> 2. ok to keep the term term "heap" in phase descriptions that apply to both heap and AO/CO tables?
Right. I believe those terms are mostly "heap_tuples_*" and "heap_blks_*". We did not rename those for AO/CO tables because we want to keep the same schema as the upstream "pg_stat_progre_<command>" tables. Under the AO/CO context. "heap_tuples_*" means just tuples or live tuples. "heap_blks_*" means equivalent number of heap blocks of given size in bytes.

> 3. do we support progress on both partitioned and non-partitioned tables?  and do we support progress in the same way for both?
Yes we do support partitioned and non-partitioned tables in same way. Specifically, right now the views show only stats of individual partitions, the parent table don't aggregate progress of its children.

> 4. would a user invoke pg_checksums directly, or indirectly via a utility command (like gprecover seg and basebackup)?
`pg_checksums` I believe is a standalone utility. We currently don't invoke it via other utility command, or vice versa. I don't know if we have any plan to integrate it though. @greenplum-db/cluster-management may have more context on that front.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md (hunk 2 above)
Comment (lisakowen) [reply gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md]: remove pg_checksums section and "with some caveats".

File: gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md (hunk 5 above)
Comment (lisakowen) [reply gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md]: added the requested text to that paragraph.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+| `acquiring inherited sample rows` | The command is currently scanning child tables to obtain sample rows. Columns `child_tables_total`, `child_tables_done`, and `current_child_table_relid` contain the progress information for this phase. |
+| `computing statistics` | The command is computing statistics from the sample rows obtained during the table scan. |
+| `computing extended statistics` | The command is computing extended statistics from the sample rows obtained during the table scan. |
+| `finalizing analyze` | The command is updating `pg_class`. When this phase is completed, `ANALYZE` ends. |
+
+
+## <a id="cluster_progress"></a>CLUSTER and VACUUM FULL Progress Reporting
+
+The [gp_stat_progress_cluster](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_cluster) system view reports the progress of running `CLUSTER`, `clusterdb`, and `VACUUM FULL` operations. (`VACUUM FULL` is similar to `CLUSTER` in that Greenplum performs a re-write of the table.) The view displays a row per segment instance that is currently servicing any of the mentioned commands.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md]: i qualified VACUUM FULL for heap tables only.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md (hunk 8 above)
Comment (lisakowen) [reply gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:83]: thanks!

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md (hunk 14 above)
Comment (lisakowen) [reply gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md]: thank you for filling in the XXX, @l-wang !

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md [hunk 16]
```diff
@@ -501,6 +513,136 @@ The `gp_stat_activity` view is a cluster-wide view that displays the [`pg_stat_a
       </div>
     </div>
 
+### <a id="gp_stat_progress_analyze"></a>gp_stat_progress_analyze
+
+The `gp_stat_progress_analyze` view is a cluster-wide view that displays the [pg_stat_progress_analyze](https://www.postgresql.org/docs/15/progress-reporting.html#ANALYZE-PROGRESS-REPORTING) information from every primary segment for all currently-running `ANALYZE` operations.
+
+The `gp_stat_progress_analyze_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_analyze`, and shares the same schema as this view.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md]: ok, i removed the ", and shares the same schema as this view." statements from the progress reporting views.  the description of the gp_segment_id in all of the progress views states "Unique identifier of a segment \(or coordinator\) instance. (This column is not present in the `gp_stat_progress_XXX_summary` view.)".  is this sufficient to address your comments in this area @l-wang ?

**Comment by lisakowen** [comment]
thank you for reviewing @Chibin and @l-wang !  i made most of your requested edits and re-pushed the review site - you may need to refresh.  only one outstanding question on my end, sent previously.

File: gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md [hunk 17]
```diff
@@ -16,14 +16,12 @@ clusterdb -V | --version
 
 ## <a id="section3"></a>Description 
 
-XXX To cluster a table means to physically reorder a table on disk according to an index so that index scan operations can access data on disk in a somewhat sequential order, thereby improving index seek performance for queries that use that index.
+To cluster a table means to physically reorder a table on disk according to an index. Clustering is a one-time operation: when the table is subsequently updated, the changes are not clustered. That is, no attempt is made to store new or updated rows according to their index order.
```
Comment (l-wang) [inline gpdb-doc/markdown/utility_guide/ref/clusterdb.html.md:19]: Can we still mention that CLUSTER helps improving index seek performance for queries that use that index?

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 18]
```diff
@@ -108,13 +108,13 @@ Greenplum Database skips several phases because it does not support concurrent i
...
 
-For each active vacuum operation, the `gp_stat_progress_vacuum_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_vacuum`, and shares the same schema as this view.
+For each active vacuum operation, the `gp_stat_progress_vacuum_summary` view aggregates across the Greenplum Database cluster the metrics reported by `gp_stat_progress_vacuum`.
 
-(Regular `VACUUM` modifies the table in place. `VACUUM FULL` rewrites the table as does a `CLUSTER` operation. For information about progress reporting for `VACUUM FULL`, see [CLUSTER and VACUUM FULL Progress Reporting](#cluster_progress).)
+(Regular `VACUUM` modifies the table in place. `VACUUM FULL` on a heap table rewrites the table as does a `CLUSTER` operation. For information about progress reporting for `VACUUM FULL` on a heap table, see [CLUSTER and VACUUM FULL Progress Reporting](#cluster_progress).)
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:115]: "Regular `VACUUM` modifies the table in place" : this is also for heap table only.

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 19]
```diff
@@ -108,13 +108,13 @@ Greenplum Database skips several phases because it does not support concurrent i
...
 
-In Greenplum Database, an AO/CO table vacuum behaves differently than a heap table vacuum. Because Greenplum stores the logical EOF for each segment file, it does not need to scan physical blocks after the logical EOF, so Greenplum can truncate them. Due to this difference, Greenplum reports vacuum progress on heap and AO/CO tables using different phases.
+In Greenplum Database, an AO/CO table vacuum behaves differently than a heap table vacuum. Because Greenplum stores the logical EOF for each segment file, it does not need to scan physical blocks after the logical EOF, so Greenplum can truncate them. Greenplum always rewrites data into new segment files to get rid of dead tuples for AO/CO tables, and so performs the same operations for both `VACUUM FULL` and `VACUUM`. Due to these difference, Greenplum reports vacuum progress on heap and AO/CO tables using different phases.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:117]: s/`and so`/`and`/ ?

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 20]
```diff
@@ -138,16 +138,16 @@ The table below describes how to interpret the *AO/CO table* phase-specific info
...
+| `append-optimized post-cleanup` | `VACUUM` is performing post-cleanup of the AO/CO table; this includes recycling old segments that are no longer visible after the "compact" phase, and truncating all live segment files to their logcial EOFs. Use `heap_blks_vacuumed` and `index_vacuum_count` to monitor the progress. |
 
 
 ## <a id="basebackup_progress"></a>Base Backup Progress Reporting
 
-The [gp_stat_progress_basebackup](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_basebackup) system view reports the progress of running base backup operations, as is performed by `gpcoverseg`. The view displays a row per segment instance that is currently servicing replication commands.
+The [gp_stat_progress_basebackup](../../ref_guide/system_catalogs/catalog_ref-views.html#gp_stat_progress_basebackup) system view reports the progress of running base backup operations, as is performed by `gprecoverseg`. The view displays a row per segment instance that is currently servicing replication commands.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:148]: s/`gprecoverseg`/`gprecoverseg -F`/ where `-F` stands for full recovery.

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md [hunk 21]
```diff
@@ -567,8 +567,8 @@ The `gp_stat_progress_cluster_summary` view aggregates across the Greenplum Data
...
 | `cluster_index_relid` | oid | If the table is being scanned using an index, this is the object identifier of the index being used; otherwise, it is zero. This field is not applicable to AO/CO tables. |
-| `heap_tuples_scanned` | bigint | Number of heap tuples scanned. This counter only advances when the phase is `seq scanning append-optimized` or `seq scanning heap`, `index scanning heap`, or `writing new heap`. For AO/CO tables, Greenplum converts byte size into equivalent heap blocks in size. |
-| `heap_tuples_written` | bigint | Number of heap tuples written. This counter only advances when the phase is `seq scanning heap`, `index scanning heap`, `writing new append-optimized`, or `writing new heap`. |
+| `heap_tuples_scanned` | bigint | Number of live tuples scanned. This counter only advances when the phase is `seq scanning append-optimized` or `seq scanning heap`, `index scanning heap`, or `writing new heap`. For AO/CO tables, Greenplum converts byte size into equivalent heap blocks in size. |
```
Comment (l-wang) [inline gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:570]: For heap table, `heap_tuples_scanned` records the number of tuples scanned, including both live and dead tuples.
For AO table, `heap_tuples_scanned` records the number of live tuples scanned, excluding the dead tuples.

**Comment by lisakowen** [comment]
@Chibin - re: your question about "greenplum database" vs just "greenplum" ...  the docs use these terms interchangeably.  i think we typically initially refer to Greenplum Database on a page, and use one or the other for later references.

**Comment by l-wang** [comment]
> QUESTIONS:
> 
> are there any limitations or known issues to call out?

1. Would it be worth mentionning progress phases and interpretations of attributes are slightly different for AO/AOCO tables from the heap tables for some commands? The commands are CREATE INDEX/REINDEX, VACUUM, CLUSTER. Maybe ask users refer to individual commands for details? I don't think this is limitation or "issue" though.
2. The progress views for ANALYZE a partitioned table shows progress of each individual leaf partitions, and no holistic progress of all partitions is available at the moment. This behavior is same as upstream Postgres.
3. CLUSTER a partitioned table is not supported. This is on par with Postgres 12. Postgres 15 does support it. (I just realized this is a regression from 6X. Let's document this for now and I will confirm if we want to re-enable it)

**Comment by lisakowen** [comment]
@l-wang , thank you for the quick turnaround reviewing my commits.

question:  the docs use AO/CO in many places for this feature.  should these be replaced with AO/AOCO?

File: gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md [hunk 22]
```diff
@@ -174,9 +174,11 @@ EOF
...
-## <a id="limitations"></a>Limitations and Known Issues
+## <a id="consider"></a>Additional Considerations
 
-Progress reporting in Greenplum Database has these known issues and limitations:
+Take note of these additional considerations related to progress reporting in Greenplum Database:
 
-- ANY???
+- Progress phases and their interpretation of attributes are slightly different for AO/AOCO tables and heap tables for the `CREATE INDEX/REINDEX`, `VACUUM`, and `CLUSTER` operations. Refer to the individual command topics for more information.
+- The progress views for an `ANALYZE` operation on a partitioned table display the progress of each individual leaf partition. Greenplum does not currently provide a measure of the combined progress of all partitions.
+- Greenplum does not support the `CLUSTER` operation on a partitioned table.
```
Comment (l-wang) [inline gpdb-doc/markdown/admin_guide/managing/progress_reporting.html.md:183]: Thanks for adding this. I just created a story in backlog to enable the same. Hopefully we can remove this one soon.

**Comment by l-wang** [comment]
> @l-wang , thank you for the quick turnaround reviewing my commits.

No problem! 

> question: the docs use AO/CO in many places for this feature. should these be replaced with AO/AOCO?

I actually have the same question when initially looking at the PR. I don't know if we've established a convention, if we haven't, it's better to decided on one and use the same name consistently across all 7X documentations, not just the doc pages for this PR. 

Personally I perfer "append-optimized" tables over "AO/AOCO" or "AO/CO" because I like full names over acronyms. Although sometimes it becomes mouthful. I also prefer "ao_row" and "ao_column" if need to refer to exact access methods since those are the actual access method names we store in the catalog. Historically we also have names like append-only, AO, CO, AOCO, AOCS, and I'm sure people have different preferences. I'm cool with whatever name we decided to use as long as we keep consistency. @ashwinstar @soumyadeep2007 @dpandhi-git @shivzone  ideas?

**Review by l-wang — APPROVED** [review]
Content look good to me! Thanks Lisa!

I have reviewed all the git diffs. Due to VPN problem I was not able to check the format through the internal web page, so it would be nice if someone else can also check the web page before publishing.

**Comment by lisakowen** [comment]
thank you, @l-wang !  i've created a new story for "use consistent terminology for append-optimized tables", will follow further discussion here.

**Review by VMWareJennyBeth — APPROVED** [review]

**Review by VMWareJennyBeth — APPROVED** [review]

**Review by VMWareJennyBeth — APPROVED** [review]

**Review by VMWareJennyBeth — APPROVED** [review]

**Pull request merged**